src/client/dfuse/ops/lookup.c

```diff
@@ -157,155 +152,50 @@ check_for_uns_ep(struct dfuse_projection_info *fs_handle,
 	if (dattr.da_type != DAOS_PROP_CO_LAYOUT_POSIX)
 		return ENOTSUP;
 
-	D_MUTEX_LOCK(&fs_handle->dpi_info->di_lock);
-
 	/* Search the currently connect dfp list, if one matches then use that,
 	 * otherwise allocate a new one.
 	 */
-	d_list_for_each_entry(dfpi, &fs_handle->dpi_info->di_dfp_list,
-			      dfp_list) {
-		DFUSE_TRA_DEBUG(ie, "Checking dfp %p", dfpi);
-
-		if (uuid_compare(dattr.da_puuid, dfpi->dfp_pool) != 0)
-			continue;
-
-		DFUSE_TRA_DEBUG(ie, "Reusing dfp %p", dfpi);
-		dfp = dfpi;
-		break;
-	}
-
-	if (!dfp) {
-		D_ALLOC_PTR(dfp);
-		if (dfp == NULL)
-			D_GOTO(out_err, ret = ENOMEM);
-
-		DFUSE_TRA_UP(dfp, ie->ie_dfs->dfs_dfp, "dfp");
-		D_INIT_LIST_HEAD(&dfp->dfp_dfs_list);
-		d_list_add(&dfp->dfp_list, &fs_handle->dpi_info->di_dfp_list);
-		new_pool = true;
-
-		uuid_copy(dfp->dfp_pool, dattr.da_puuid);
-
-		/* Connect to DAOS pool */
-		rc = daos_pool_connect(dfp->dfp_pool,
-				       fs_handle->dpi_info->di_group,
-				       DAOS_PC_RW,
-				       &dfp->dfp_poh, &dfp->dfp_pool_info,
-				       NULL);
-		if (rc != -DER_SUCCESS) {
-			if (rc == -DER_NO_PERM)
-				DFUSE_TRA_DEBUG(ie,
-						"daos_pool_connect() failed, "
-						DF_RC"\n", DP_RC(rc));
-			else
-				DFUSE_TRA_WARNING(ie,
-						  "daos_pool_connect() failed, "
-						  DF_RC"\n", DP_RC(rc));
-			D_GOTO(out_err, ret = daos_der2errno(rc));
-		}
-	}
-
-	d_list_for_each_entry(dfsi, &dfp->dfp_dfs_list,	dfs_list) {
-		if (uuid_compare(dattr.da_cuuid, dfsi->dfs_cont) != 0)
-			continue;
-
-		DFUSE_TRA_DEBUG(ie, "Reusing dfs %p", dfsi);
-		dfs = dfsi;
-		break;
-	}
 
-	if (!dfs) {
-		D_ALLOC_PTR(dfs);
-		if (dfs == NULL)
-			D_GOTO(out_pool, ret = ENOMEM);
-
-		dfs->dfs_ops = ie->ie_dfs->dfs_ops;
-		DFUSE_TRA_UP(dfs, dfp, "dfs");
-		uuid_copy(dfs->dfs_cont, dattr.da_cuuid);
-		d_list_add(&dfs->dfs_list, &dfp->dfp_dfs_list);
-		new_cont = true;
-
-		dfuse_dfs_init(dfs, ie->ie_dfs);
-
-		/* Try to open the DAOS container (the mountpoint) */
-		rc = daos_cont_open(dfp->dfp_poh, dfs->dfs_cont, DAOS_COO_RW,
-				    &dfs->dfs_coh, NULL, NULL);
-		if (rc) {
-			DFUSE_LOG_ERROR("Failed container open (%d)",
-					rc);
-			D_GOTO(out_pool, ret = daos_der2errno(rc));
-		}
+ 	rc = dfuse_pool_open(fs_handle, &dattr.da_puuid, &dfp);
+ 	if (rc != -DER_SUCCESS)
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs